executors/multiversx/filters/pendingOperationFilter.go

```diff
@@ -0,0 +1,221 @@
+	)
+
+	return filter, nil
+}
+
+func (filter *pendingOperationFilter) parseConfigs(cfg config.PendingOperationsFilterConfig) error {
+	var err error
+
+	// denied lists do not support wildcard items
+	filter.deniedEthAddresses, err = parseList(cfg.DeniedEthAddresses, wildcardString)
+	if err != nil {
+		return fmt.Errorf("%w in list DeniedEthAddresses", err)
+	}
+
+	filter.deniedMvxAddresses, err = parseList(cfg.DeniedMvxAddresses, wildcardString)
+	if err != nil {
+		return fmt.Errorf("%w in list DeniedMvxAddresses", err)
+	}
+
+	filter.deniedTokens, err = parseList(cfg.DeniedTokens, wildcardString)
+	if err != nil {
+		return fmt.Errorf("%w in list DeniedTokens", err)
+	}
+
+	// allowed lists do not support empty items
+	filter.allowedEthAddresses, err = parseList(cfg.AllowedEthAddresses, emptyString)
+	if err != nil {
+		return fmt.Errorf("%w in list AllowedEthAddresses", err)
+	}
+
+	filter.allowedMvxAddresses, err = parseList(cfg.AllowedMvxAddresses, emptyString)
+	if err != nil {
+		return fmt.Errorf("%w in list AllowedMvxAddresses", err)
+	}
+
+	filter.allowedTokens, err = parseList(cfg.AllowedTokens, emptyString)
+	if err != nil {
+		return fmt.Errorf("%w in list AllowedTokens", err)
+	}
+
+	return nil
+}
+
+func parseList(list []string, unsupportedMarker string) ([]string, error) {
+	newList := make([]string, 0, len(list))
+	for index, item := range list {
+		item = strings.ToLower(item)
+		item = strings.Trim(item, "\r\n \t")
+		if item == unsupportedMarker {
+			return nil, fmt.Errorf("%w %s on item at index %d", errUnsupportedMarker, unsupportedMarker, index)
+		}
+
+		newList = append(newList, item)
+	}
+
+	return newList, nil
+}
+
+func (filter *pendingOperationFilter) checkLists() error {
+	err := filter.checkEthList(filter.allowedEthAddresses)
+	if err != nil {
+		return fmt.Errorf("%w in list AllowedEthAddresses", err)
+	}
+
+	err = filter.checkEthList(filter.deniedEthAddresses)
+	if err != nil {
+		return fmt.Errorf("%w in list DeniedEthAddresses", err)
+	}
+
+	err = filter.checkMvxList(filter.allowedMvxAddresses)
+	if err != nil {
+		return fmt.Errorf("%w in list AllowedMvxAddresses", err)
+	}
+
+	err = filter.checkMvxList(filter.deniedMvxAddresses)
+	if err != nil {
+		return fmt.Errorf("%w in list DeniedMvxAddresses", err)
+	}
+
+	return nil
+}
+
+func (filter *pendingOperationFilter) checkEthList(list []string) error {
+	for index, item := range list {
+		if item == wildcardString {
+			continue
+		}
+
+		if !strings.HasPrefix(item, ethAddressPrefix) {
+			return fmt.Errorf("%w (%s) on item at index %d", errMissingEthPrefix, ethAddressPrefix, index)
+		}
+	}
+
+	return nil
+}
+
+func (filter *pendingOperationFilter) checkMvxList(list []string) error {
+	for index, item := range list {
+		if item == wildcardString {
+			continue
+		}
+
+		_, err := data.NewAddressFromBech32String(item)
+		if err != nil {
+			return fmt.Errorf("%w on item at index %d", err, index)
+		}
+	}
+
+	return nil
+}
```

Comment:
optional, not a strong opinion, could be reduced to:
```go
func (filter *pendingOperationFilter) checkLists() error {
	err := filter.checkList(filter.allowedEthAddresses, isEthItemValid)
	if err != nil {
		return fmt.Errorf("%w in list AllowedEthAddresses", err)
	}

	err = filter.checkList(filter.deniedEthAddresses, isEthItemValid)
	if err != nil {
		return fmt.Errorf("%w in list DeniedEthAddresses", err)
	}

	err = filter.checkList(filter.allowedMvxAddresses, isMvxItemValid)
	if err != nil {
		return fmt.Errorf("%w in list AllowedMvxAddresses", err)
	}

	err = filter.checkList(filter.deniedMvxAddresses, isMvxItemValid)
	if err != nil {
		return fmt.Errorf("%w in list DeniedMvxAddresses", err)
	}

	return nil
}

func isMvxItemValid(item string) bool {
	_, errNewAddr := data.NewAddressFromBech32String(item)
	return errNewAddr == nil
}

func isEthItemValid(item string) bool {
	return !strings.HasPrefix(item, ethAddressPrefix)
}

func (filter *pendingOperationFilter) checkList(list []string, isItemValid func(item string) bool) error {
	for index, item := range list {
		if item == wildcardString {
			continue
		}

		if !isItemValid(item) {
			return fmt.Errorf("%w on item at index %d", errInvalidItem, index)
		}
	}

	return nil
}
```

Reply:
a little bit reticent on using function pointers, but agree with this solution. Changed